include/seqan3/alignment/aligned_sequence/aligned_sequence_concept.hpp

```diff
@@ -272,4 +285,92 @@ inline typename seq_type::iterator erase_gap(seq_type & seq, typename seq_type::
 }
 //!\}
 
+namespace detail
+{
+
+/*!\brief               Create the formatted alignment output and add it to the provided debug_stream.
+ * \ingroup             aligned_sequence
+ * \tparam alignment_t  The type of the alignment, must satisfy tuple_like_concept.
+ * \tparam idx          An index sequence.
+ * \param[in] stream    The output stream that receives the formatted alignment.
+ * \param[in] align     The alignment that shall be streamed.
+ */
+template<tuple_like_concept alignment_t, size_t ...idx>
+void stream_alignment(debug_stream_type & stream, alignment_t const & align, std::index_sequence<idx...> const & /**/)
+{
+    size_t const alignment_size = std::get<0>(align).size();
+
+    // split alignment into blocks of length 50 and loop over parts
+    for (size_t begin_pos = 0; begin_pos < alignment_size; begin_pos += 50)
+    {
+        size_t const end_pos = std::min(begin_pos + 50, alignment_size);
+
+        // write header line
+        if (begin_pos != 0)
+            stream << '\n';
+
+        stream << std::setw(7) << begin_pos << ' ';
+        for (size_t rel_pos = 1; rel_pos <= (end_pos - 1) % 50 + 1; ++rel_pos)
```

Comment:
```suggestion
        for (size_t pos = begin_pos + 1; pos <= end_pos; ++pos)
```

You already do a modulo in the loop you don't need to start at position 1, because begin_pos is a multiple of 50,

Let's relax `rel_pos = 1 + k` (where `k in [0, 50)` is the current iteration) to

```
rel_pos' = begin_pos + 1 + k
```

Then

```
        rel_pos' % 10 == 0
<=>     ( begin_pos + 1 + k ) % 10 == 0
<=>     ( begin_pos % 10 + (1 + k) % 10 ) == 0 // begin_pos % 50 is zero, therefore also begin_pos % 10
<=>     ( (1 + k) % 10 ) == 0
<=>     ( rel_pos % 10 ) == 0  // the original condition holds
```
The same holds for `% 5`.